src/spikeinterface/core/recording_tools.py

```diff
@@ -110,8 +110,7 @@ def write_binary_recording(
         file_path_list = [add_suffix(file_path, ["raw", "bin", "dat"]) for file_path in file_path_list]
 
     dtype = dtype if dtype is not None else recording.get_dtype()
-    cast_unsigned = False
-    if auto_cast_uint:
+    if auto_cast_uint:  # TODO should we deprecate this given that we have `unsigned_to_signed`?
```

Comment:
Yes!

Reply:
@h-mayorquin can you add it to this PR?

Reply:
Done.

Reply:
I also improved the docstrings a little bit.

Reply:
The docstring looks nicer to me :)